camel-core/src/main/java/org/apache/camel/processor/aggregate/AggregateProcessor.java

```diff
@@ -1223,68 +1223,77 @@ public void run() {
-                } else {
-                    LOG.debug("Loading aggregated exchange with id: {} to be recovered.", exchangeId);
-                    Exchange exchange = recoverable.recover(camelContext, exchangeId);
-                    if (exchange != null) {
-                        // get the correlation key
-                        String key = exchange.getProperty(Exchange.AGGREGATED_CORRELATION_KEY, String.class);
-                        // and mark it as redelivered
-                        exchange.getIn().setHeader(Exchange.REDELIVERED, Boolean.TRUE);
-
-                        // get the current redelivery data
-                        RedeliveryData data = redeliveryState.get(exchange.getExchangeId());
-
-                        // if we are exhausted, then move to dead letter channel
-                        if (data != null && recoverable.getMaximumRedeliveries() > 0 && data.redeliveryCounter >= recoverable.getMaximumRedeliveries()) {
-                            LOG.warn("The recovered exchange is exhausted after " + recoverable.getMaximumRedeliveries()
-                                    + " attempts, will now be moved to dead letter channel: " + recoverable.getDeadLetterUri());
+                                // handle if failed
+                                if (exchange.getException() != null) {
+                                    getExceptionHandler().handleException("Failed to move recovered Exchange to dead letter channel: " + recoverable.getDeadLetterUri(), exchange.getException());
+                                } else {
+                                    // it was ok, so confirm after it has been moved to dead letter channel, so we wont recover it again
+                                    recoverable.confirm(camelContext, exchangeId);
+                                }
+                            } else {
+                                // update current redelivery state
+                                if (data == null) {
+                                    // create new data
+                                    data = new RedeliveryData();
+                                    redeliveryState.put(exchange.getExchangeId(), data);
+                                }
+                                data.redeliveryCounter++;
 
-                            // send to DLC
-                            try {
                                 // set redelivery counter
                                 exchange.getIn().setHeader(Exchange.REDELIVERY_COUNTER, data.redeliveryCounter);
-                                exchange.getIn().setHeader(Exchange.REDELIVERY_EXHAUSTED, Boolean.TRUE);
-                                deadLetterProducerTemplate.send(recoverable.getDeadLetterUri(), exchange);
-                            } catch (Throwable e) {
-                                exchange.setException(e);
-                            }
+                                if (recoverable.getMaximumRedeliveries() > 0) {
```

Comment:
yes, probable formatting confusion on my side. LG.
I would suggest running full core unit tests. if there is not an issue, i guess this PR may go into further releases but i guess still this is not a full resolution to CAMEL-8010 issue?